staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -115,6 +115,8 @@ func (c *crConverter) ConvertFieldLabel(gvk schema.GroupVersionKind, label, valu
 		return label, value, nil
 	case !c.clusterScoped && label == "metadata.namespace":
 		return label, value, nil
+	case gvk.Kind == "Pod" && label == "spec.nodeName":
```

Comment:
This is in fact an unrelated change that should be removed. I did it when trying to plug `virtual-kubelet` directly to KCP.
But obviously this should be fixed more widely in the future by enabling all possible fields in the filters for CRDs according to the CRD schema as we discussed.